### PR TITLE
[VBLOCKS-3800] feat: cold transfer

### PR DESCRIPTION
**Pull request by kpchoy**
Steps
- agent1 calls customer
- agent1 forwards call to agent2
- agent1 immediately drops from call after "forward" button is clicked
- agent2 picks up call and continues conversation with customer

Screenshot 2024-12-04 at 2 12 46 PM

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: src/common/routes.js [hunk 1]
```diff
@@ -109,32 +109,41 @@ export const conferenceEventsHandler = async (req, res) => {
       .conferences(ConferenceSid)
       .participants.list();
 
-    const isParticipantLeave = StatusCallbackEvent === 'participant-leave';
-
     // Send the conference sid and the other participants' callSids to any client leg.
     // The sids will be used to update the participant resource such as putting it on hold.
-    participants.forEach((currentParticipant) => {
+    participants.forEach(async (currentParticipant) => {
```
Comment (mhuynh5757) [inline src/common/routes.js:114]: Is there any UI logic that happens around this? Should we map each async function to a Promise and await `Promise.all`?

File: src/common/routes.js [hunk 2]
```diff
@@ -109,32 +109,41 @@ export const conferenceEventsHandler = async (req, res) => {
...
-                  muted: isParticipantLeave ? Muted : otherParticipant.muted,
-                  hold: isParticipantLeave ? Hold : otherParticipant.hold,
-                  remove: isParticipantLeave,
-                }),
-              });
-          });
+        if (StatusCallbackEvent === 'participant-leave') {
+          await client
+            .calls(currentParticipant.callSid)
+            .userDefinedMessages.create({
```
Comment (mhuynh5757) [inline src/common/routes.js:119]: Nitpick:
```suggestion
            .userDefinedMessages
            .create({
```

File: src/common/routes.js [hunk 3]
```diff
@@ -109,32 +109,41 @@ export const conferenceEventsHandler = async (req, res) => {
...
+                remove: true,
+              }),
+            });
+        } else {
+          participants
+            .filter((p) => p.callSid !== currentParticipant.callSid)
+            .forEach(async (otherParticipant) => {
+              await client
+                .calls(currentParticipant.callSid)
+                .userDefinedMessages.create({
```
Comment (mhuynh5757) [inline src/common/routes.js:135]: Same formatting nitpick.

File: src/common/routes.js (hunk 1 above)
Comment (kpchoy) [reply src/common/routes.js:114]: @mhuynh5757 Yes there is UI logic that happens, but the UI logic happens with each `UserDefinedMessage`. Even if we await all these messages in `Promise.all`, the `UserDefinedMessage(s)` still send as individual messages. The API does not return the messages. Or do you think we should handle these messages differently?

Comment (mhuynh5757) [reply src/common/routes.js:114]: As in, the UI changes when receiving the messages? Ah I see, this is the `routes.js` file so this is on the server side. I think it's good as-is then.

**Review by mhuynh5757 — APPROVED** [review]

File: src/common/routes.js [hunk 4]
```diff
@@ -109,32 +109,43 @@ export const conferenceEventsHandler = async (req, res) => {
...
-                  label: isParticipantLeave
-                    ? ParticipantLabel
-                    : otherParticipant.label,
-                  muted: isParticipantLeave ? Muted : otherParticipant.muted,
-                  hold: isParticipantLeave ? Hold : otherParticipant.hold,
-                  remove: isParticipantLeave,
-                }),
-              });
-          });
+        if (StatusCallbackEvent === 'participant-leave') {
```
Comment (charliesantos) [inline src/common/routes.js:116]: Add a comment above the `if` and `else` statement and describe what these blocks are doing. See lines 112-113 above for an example.

**Review by charliesantos — CHANGES_REQUESTED** [review]

**Review by charliesantos — APPROVED** [review]

**Pull request merged**